Error parsing test runner configuration file

Since new upgrade to BRJS v1.3.0 when running tests I get this error.

```
INFO: HV000001: Hibernate Validator 5.0.1.Final
The app 'salestrader' is either missing BRJS jar(s), contains BRJS jar(s) it shouldn't or the BRJS jar(s) are outdated. You should delete all jars prefixed with 'brjs-' in the WEB-INF/lib directory and copy in all jars contained in 'sdk/libs/java/application'.
Error:
  Error parsing test runner configuration file '/home/mvanko/Projects/salestrader/conf/test-runner.conf'.; null

Caused By:
  null

Stack Trace:
org.bladerunnerjs.api.model.exception.command.CommandOperationException: Error parsing test runner configuration file '/home/mvanko/Projects/salestrader/conf/test-runner.conf'.; null
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunnerController.run(TestRunnerController.java:76)
	at org.bladerunnerjs.legacy.command.test.TestCommand.doCommand(TestCommand.java:59)
	at org.bladerunnerjs.api.plugin.JSAPArgsParsingCommandPlugin.doCommand(JSAPArgsParsingCommandPlugin.java:86)
	at org.bladerunnerjs.utility.CommandRunner.run(CommandRunner.java:25)
	at org.bladerunnerjs.api.BRJS.runCommand(BRJS.java:430)
	at org.bladerunnerjs.utility.UserCommandRunner.doRunCommand(UserCommandRunner.java:26)
	at org.bladerunnerjs.utility.UserCommandRunner.run(UserCommandRunner.java:18)
	at org.bladerunnerjs.api.BRJS.runUserCommand(BRJS.java:435)
	at org.bladerunnerjs.runner.CommandRunner.run(CommandRunner.java:126)
	at org.bladerunnerjs.runner.CommandRunner.run(CommandRunner.java:104)
	at org.bladerunnerjs.runner.CommandRunner.main(CommandRunner.java:70)
Caused by: java.lang.NullPointerException
	at org.bladerunnerjs.legacy.conf.TestRunnerConfiguration.addDefaultBrowserIfNoBrowserDefinedInArgumentsAndArgumentDoesntAskForNone(TestRunnerConfiguration.java:162)
	at org.bladerunnerjs.legacy.conf.TestRunnerConfiguration.getBrowsers(TestRunnerConfiguration.java:61)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.getBrowsers(TestRunner.java:126)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.<init>(TestRunner.java:99)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunnerController.run(TestRunnerController.java:72)
	... 10 more

org.bladerunnerjs.api.model.exception.command.CommandOperationException: Error parsing test runner configuration file '/home/mvanko/Projects/salestrader/conf/test-runner.conf'.; null
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunnerController.run(TestRunnerController.java:76)
	at org.bladerunnerjs.legacy.command.test.TestCommand.doCommand(TestCommand.java:59)
	at org.bladerunnerjs.api.plugin.JSAPArgsParsingCommandPlugin.doCommand(JSAPArgsParsingCommandPlugin.java:86)
	at org.bladerunnerjs.utility.CommandRunner.run(CommandRunner.java:25)
	at org.bladerunnerjs.api.BRJS.runCommand(BRJS.java:430)
	at org.bladerunnerjs.utility.UserCommandRunner.doRunCommand(UserCommandRunner.java:26)
	at org.bladerunnerjs.utility.UserCommandRunner.run(UserCommandRunner.java:18)
	at org.bladerunnerjs.api.BRJS.runUserCommand(BRJS.java:435)
	at org.bladerunnerjs.runner.CommandRunner.run(CommandRunner.java:126)
	at org.bladerunnerjs.runner.CommandRunner.run(CommandRunner.java:104)
	at org.bladerunnerjs.runner.CommandRunner.main(CommandRunner.java:70)
Caused by: java.lang.NullPointerException
	at org.bladerunnerjs.legacy.conf.TestRunnerConfiguration.addDefaultBrowserIfNoBrowserDefinedInArgumentsAndArgumentDoesntAskForNone(TestRunnerConfiguration.java:162)
	at org.bladerunnerjs.legacy.conf.TestRunnerConfiguration.getBrowsers(TestRunnerConfiguration.java:61)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.getBrowsers(TestRunner.java:126)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.<init>(TestRunner.java:99)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunnerController.run(TestRunnerController.java:72)
	... 10 more
```

This doesn't happen on earlier versions and everything is running fine.

Running on a Linux - Fedora 22
AFAIK it is running fine on a windows machine

<!---
@huboard:{"order":1595.5,"milestone_order":1630,"custom_state":""}
-->

> i cloned ct, develop branch, gradle assemble followed by brjs test libs/javascript/ct-alerts. using a centos 5 machine

@thecapdan, this worked provided I ran `brjs test-server -b none` first and then manually connected a browser, but failed when I tried to run it without manually hooking up a browser with this error:

```
ERROR: java.io.IOException: incorrect number of browser instances connected to the server: expected 1 but there were actually 0 instances connected.
Error:
  Error running tests.; incorrect number of browser instances connected to the server: expected 1 but there were actually 0 instances connected.

Caused By:
  incorrect number of browser instances connected to the server: expected 1 but there were actually 0 instances connected.

Stack Trace:
org.bladerunnerjs.api.model.exception.command.CommandOperationException: Error running tests.; incorrect number of browser instances connected to the server: expected 1 but there were actually 0 instances connected.
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunnerController.run(TestRunnerController.java:126)
	at org.bladerunnerjs.legacy.command.test.TestCommand.doCommand(TestCommand.java:54)
	at org.bladerunnerjs.utility.CommandRunner.run(CommandRunner.java:25)
	at org.bladerunnerjs.api.BRJS.runCommand(BRJS.java:428)
	at org.bladerunnerjs.utility.UserCommandRunner.doRunCommand(UserCommandRunner.java:34)
	at org.bladerunnerjs.utility.UserCommandRunner.run(UserCommandRunner.java:26)
	at org.bladerunnerjs.api.BRJS.runUserCommand(BRJS.java:433)
	at org.bladerunnerjs.runner.CommandRunner.run(CommandRunner.java:119)
	at org.bladerunnerjs.runner.CommandRunner.run(CommandRunner.java:123)
	at org.bladerunnerjs.runner.CommandRunner.main(CommandRunner.java:71)
Caused by: java.io.IOException: incorrect number of browser instances connected to the server: expected 1 but there were actually 0 instances connected.
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.waitForServer(TestRunner.java:562)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.startBrowserProcesses(TestRunner.java:498)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.startServer(TestRunner.java:322)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.runTests(TestRunner.java:157)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunnerController.run(TestRunnerController.java:121)
	... 9 more

org.bladerunnerjs.api.model.exception.command.CommandOperationException: Error running tests.; incorrect number of browser instances connected to the server: expected 1 but there were actually 0 instances connected.
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunnerController.run(TestRunnerController.java:126)
	at org.bladerunnerjs.legacy.command.test.TestCommand.doCommand(TestCommand.java:54)
	at org.bladerunnerjs.utility.CommandRunner.run(CommandRunner.java:25)
	at org.bladerunnerjs.api.BRJS.runCommand(BRJS.java:428)
	at org.bladerunnerjs.utility.UserCommandRunner.doRunCommand(UserCommandRunner.java:34)
	at org.bladerunnerjs.utility.UserCommandRunner.run(UserCommandRunner.java:26)
	at org.bladerunnerjs.api.BRJS.runUserCommand(BRJS.java:433)
	at org.bladerunnerjs.runner.CommandRunner.run(CommandRunner.java:119)
	at org.bladerunnerjs.runner.CommandRunner.run(CommandRunner.java:123)
	at org.bladerunnerjs.runner.CommandRunner.main(CommandRunner.java:71)
Caused by: java.io.IOException: incorrect number of browser instances connected to the server: expected 1 but there were actually 0 instances connected.
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.waitForServer(TestRunner.java:562)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.startBrowserProcesses(TestRunner.java:498)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.startServer(TestRunner.java:322)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunner.runTests(TestRunner.java:157)
	at org.bladerunnerjs.legacy.command.test.testrunner.TestRunnerController.run(TestRunnerController.java:121)
	... 9 more
```

but this is only because the CT build doesn't support 64 Linux browsers (the BRJS build does), and is not the same as the errors you and have @michalvankodev have been seeing. I notice that you and  @michalvankodev are both running RedHat flavored Linux distros. Are these both 64 bit?

--- a/brjs-core/src/main/java/org/bladerunnerjs/api/spec/exception/NoBrowsersDefinedException.java
+++ b/brjs-core/src/main/java/org/bladerunnerjs/api/spec/exception/NoBrowsersDefinedException.java
@@ -17,4 +17,9 @@ public class NoBrowsersDefinedException extends Exception {
 	{
 		super("Could not find any configured browsers. Please check your test configuration file inside \n\t'" + confPath + "'");
 	}
+
+	public NoBrowsersDefinedException(String confPath, String operatingSystem)
+	{
+		super("Could not find any configured browsers for this Operating System (" + operatingSystem + "). Please check your test configuration file inside \n\t'" + confPath + "'");
+	}
 }

--- a/brjs-legacy/src/main/java/org/bladerunnerjs/legacy/conf/TestRunnerConfiguration.java
+++ b/brjs-legacy/src/main/java/org/bladerunnerjs/legacy/conf/TestRunnerConfiguration.java
@@ -67,13 +67,12 @@ public class TestRunnerConfiguration {
 	public Map<String, String> getBrowserPathsForOS() throws NoBrowsersDefinedException, IOException
 	{
 		Map<String, String> paths = null;
-		try {
-			paths = browserPaths.get(getOperatingSystem());
+		paths = browserPaths.get(getOperatingSystem());
+
+		if(paths == null){
+			throw new NoBrowsersDefinedException(getRelativeDir().getAbsolutePath(), getOperatingSystem());
 		}
-		catch (ClassCastException e) 
-		{
-			throw new NoBrowsersDefinedException(getRelativeDir().getAbsolutePath());
-		}
+
 		return paths;
 	}
 	

